Nos/Views/Components/Media/ImageButton.swift

```diff
@@ -9,25 +9,38 @@ struct ImageButton: View {
     /// Whether the image viewer is presented or not.
     @State private var isViewerPresented = false
 
+    /// The size of the image that was loaded.
+    @State private var imageSize: CGSize?
+
     var body: some View {
-        Button {
-            isViewerPresented = true
-        } label: {
-            WebImage(url: url)
-                .resizable()
-                .indicator(.activity)
-                .aspectRatio(contentMode: .fill)
-                .frame(
-                    minWidth: 0,
-                    maxWidth: .infinity,
-                    minHeight: 0,
-                    maxHeight: .infinity
-                )
-                .aspectRatio(4 / 3, contentMode: .fit)
-                .clipShape(.rect)
-        }
-        .sheet(isPresented: $isViewerPresented) {
-            ImageViewer(url: url)
+        Color.clear
+            .aspectRatio(aspectRatio, contentMode: .fit)
+            .overlay(
+                Button {
+                    isViewerPresented = true
+                } label: {
+                    WebImage(url: url)
+                        .onSuccess { image, _, _ in
+                            imageSize = image.size
+                        }
+                        .resizable()
+                        .indicator(.activity)
+                        .scaledToFill()
+                }
+            )
+            .clipShape(.rect)
+            .contentShape(.rect)
+            .sheet(isPresented: $isViewerPresented) {
+                ImageViewer(url: url)
+            }
```

Comment:
@martindsq what are your thoughts on this refactor? Instead of setting the aspect ratio twice on the WebImage and setting its frame, we can just use `Color.clear` with an aspect ratio and overlay. This is exactly what we do in SquareImage. The only reason I didn't implement this the same as SquareImage before is that I couldn't get it to work right and show the image in the right aspect ratio. I must have missed something, though, because it's working now.

Reply:
I guess this approach adds a layer with the Color.clear and the overlay on top of it. With the other approach, we had to set the aspect ratio twice. I'm not sure which I dislike less.

Reply:
I like that!

Reply:
I have used Color.clear in some cases, it makes sense to use it in SwiftUI unfortunately. If you need arguments in order to discard the other approach, I think that would be that it was dependent on the specific orden in which you set the aspect ratio. That is hard to understand, and also sounds like a hack in some way.

Reply:
@martindsq to be clear, you like the new approach where we don't have to set the frame, right?